### ISSUE TITLE
set: Warning: path component /some/path may not be valid in PATH.

### ISSUE THREAD

**Issue by jurgenhaas** [fish-shell/fish-shell#3150]
I know there have been other issues like #466 or #14 or #8 around that describe this issue and left me under the impression it should be fixed, but I'm still getting it.

I'm using Fish 2.3.0 and I have the following line in /etc/fish/config.fish:

```
set -g -x fish_user_paths /some/path
```

That path did not exist and hence I got the warning about the invalid PATH. Then I created that path and made sure it's readable by every user, still the warning appears.

As #466 talked about inheritance from a parent process I could figure out what that parent process should be and rebooted the host. Still the same warning.

Then I removed that component from the $PATH environment variable and still the warning appears.

Now I'm running out of ideas and wonder what I should be doing about this.

**Comment by faho** [comment]
The obvious thing to look for here are universal variables. $PATH is global by default, while `fish_user_paths` is universal. It's possible you set a universal variable by accident.

So, right after starting fish, run `echo $PATH; echo $fish_user_paths`.

Also if you are using OSX, inspect /etc/paths and /etc/paths.d/* for any invalid components.

**Comment by krader1961** [comment]
The behavior you're seeing is intentional. Fish is warning you that */some/path* doesn't exist and therefore you might have made a typographical mistake. When you say you "created that path" what command did you run? Note that fish only checks that the path refers to a directory. It doesn't even check the permissions of the directory.

**Comment by krader1961** [comment]
Also, you probably don't want to export (i.e., `set -x`) that variable. It's not likely to cause any problems but neither is it doing anything useful.

**Comment by jurgenhaas** [comment]
@faho the two variables have these values:

```
$PATH: /usr/local/share/vendor/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin
$fish_user_path: /usr/local/share/vendor/bin
```

I'm on Ubuntu 14.04 with this and when I do a `ls /usr/local/share/vendor/bin -lisa` I do get a result which clearly shows that this path exists. I do that for all other components in $PATH too and all the directories do exist. Just in case, I do that with copy and paste, so a typo is definitely not the reason for the problem.

@krader1961 I have created the directory with `mkdir`.

 [ ... time shift ... ]

I was digging deeper and what confused me even more was the fact that this message did not occur for each and every command I executed but only for some. When I called `svnsync help` the warning was not displayed, but when I called `svnsync --non-interactive sync svn://localhost/var/svn/d7core/` then the warning was in fact displayed. So guess what? The SVN repository at `/var/svn/d7core` was setup to sync with another SVN repository on a different host and the warning came from that remote host which had the same setup as the local one but I only had tried to fix it locally before doing it everywhere.

As soon as I created the directory on each host where it was missing, the problem has went away. Sorry for the confusion, but maybe that's gonna help someone else eventually.
